sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -348,6 +401,16 @@ Applications must make sure that the Level-Zero handles themselves aren't used s
 Practically speaking, and taking into account that SYCL runtime takes ownership of the Level-Zero handles,
 the application should not attempt further direct use of those handles.
 
+### 4.5 Interoperability buffer synchronization rules
+
+A SYCL buffer that is constructed with this interop API uses the Level Zero memory buffer for its full lifetime, and the contents of the Level Zero memory buffer are unspecified for the lifetime of the SYCL buffer. If the application modifies the contents of that Level Zero memory buffer during the lifetime of the SYCL buffer, the behavior is undefined. The initial contents of the SYCL buffer will be the initial contents of the Level Zero memory buffer at the time of the SYCL buffer's construction.
+
+The behavior of the SYCL buffer destructor depends on the Ownership flag. As with other SYCL buffers, this behavior is triggered only when the last reference count to the buffer is dropped, as described in the core SYCL specification section 4.7.2.3, "Buffer synchronization rules".
+
+* If the ownership is keep (i.e. the application retains ownership of the Level Zero memory buffer), then the SYCL buffer destructor blocks until all work in queues on the buffer have completed. The buffer's contents are written to the Level Zero memory buffer by the time the destructor completes.
+
+* If the ownership is transfer (i.e. the SYCL runtime has ownership of the Level Zero memory buffer), then the SYCL buffer destructor blocks until all work in queues on the buffer have completed. The SYCL runtime frees the Level Zero memory buffer asynchronously when it is no longer in use in queues.

```

Comment:
I see, indeed spec says that destructor is not necessarily blocking:

> A buffer can be constructed with just a size and using the default buffer allocator. ... The destructor for this type of buffer does not need to block, even if work on the buffer has not completed. 

But problem is that currently implementation is blocking in this case, even if I do set_write_back(false) explicitly. Implementation of buffer destructor is blocking in all cases now.

I propose to use the same wording for interop buffer - "does not need to block" instead of "doesn't block". And in the future if non-blocking version of the buffer will be implemented we will use it for "transfer" semantics as well.

Reply:
> I propose to use the same wording for interop buffer - "does not need to block" instead of "doesn't block".

Given that we use this wording also in the core spec, I do not object to using it in this extension spec.